task.go

```diff
@@ -192,6 +194,9 @@ func (e *Executor) RunTask(ctx context.Context, call taskfile.Call) error {
 				return &TaskRunError{t.Task, err}
 			}
 		}
+		if e.PropStatus && !e.OutOfDate {
+			e.OutOfDate = true
```

Comment:
Hi @ConX.

Unfortunately, implementing this will actually be more complicated IMO.

FYI, changing a variable here without a mutex or similar is not ideal because tasks can be run in parallel.

That said, I think the implementation would actually be different. For example, it's possible to run multiple tasks like `task foo bar baz`, even in parallel with the `-p` flag, so this would require code to check specific dependencies of each task before running each of them, instead of having a global one.

I'm not sure _how_ to proceed exactly because I haven't dived in, but I wanted to let you know my initial thought about this proposal.

Reply:
Hi @andreynering,

Thanks for checking the PR.

You are correct on both points; I didn't think of the parallel aspect of task, which is definitely one of its strengths. I will think about it myself, but please let me know if you have an approach in mind, and I could give it a try if you don't have time.